### PR TITLE
Fluent 9 in Cardboard

### PR DESCRIPTION
**Pull request by pasanchMSFT**
### Summary of changes 🔍 
Added Fluent 9 package, babel configuration and linting rules.

Added Fluent theme provider to `BaseComponent`, and also created very custom Fluent 9 themes so they could match our current themes. Current solution is planned to be temporary, and themes will probably move to what Fluent 9 offers.

Created a component with new Fluent 9 patterns and also some templates to start developing new components with Fluent 9.

image

### Testing 🧪
Open any story and take a look at styles, everything should be the same across all themes.
Open `ConsumeShell` story to take a look at a new component built with Fluent 9 styling.

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
         "generate-schema": "node .\\tools\\generate-3d-schema-types.js",
         "generate": "node ./tools/generate-component-template.js",
         "generate-legion": "node ./tools/generate-legion-template.js",
```
Comment (msnyder-msft) [inline package.json:23]: What does `generate-next` do?

File: src/Apps/Legion/Components/ConsumeShell/ConsumeShell.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                style={{ [`${gridBorderVar}`]: theme.palette.glassyBorder }}
+            >
+                <GlobeRegular fontSize={'18px'} />
+                <h3 className={classNames.headerText}>
+                    {appDataState?.targetDatabase?.databaseName}
+                </h3>
+            </div>
+            <div
+                className={classNames.leftNav}
+                style={{ [`${gridBorderVar}`]: theme.palette.glassyBorder }}
```
Comment (msnyder-msft) [inline src/Apps/Legion/Components/ConsumeShell/ConsumeShell.tsx:43]: do you need it in the special ticks? or can it just be the variable directly?

File: src/Apps/Legion/Components/ConsumeShell/ConsumeShell.types.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface IConsumeShellProps {}
+
+export type TConsumeShellClassNames =
```
Comment (msnyder-msft) [inline src/Apps/Legion/Components/ConsumeShell/ConsumeShell.types.ts:4]: I'm probably just missing it, but where does this get used?

File: src/Theming/Fluent9DarkTheme.ts [hunk 4]
```diff
@@ -0,0 +1,397 @@
+import { Theme } from '@fluentui/react-components';
+
+export const Fluent9DarkTheme: Theme = {
```
Comment (msnyder-msft) [inline src/Theming/Fluent9DarkTheme.ts:3]: so you have to define everything explicitly? it's not just the overrides?

File: src/Apps/Legion/Components/WizardShell/Internal/SaveStep/SaveStep.styles.ts [hunk 5]
```diff
@@ -9,7 +9,12 @@ const classNames = {
 // export const SAVESTEP_CLASS_NAMES = classNames;
 export const getStyles = (_props: ISaveStepStyleProps): ISaveStepStyles => {
     return {
-        root: [classNames.root],
+        root: [
+            classNames.root,
+            {
+                height: '440px'
```
Comment (msnyder-msft) [inline src/Apps/Legion/Components/WizardShell/Internal/SaveStep/SaveStep.styles.ts:15]: is this the same height assumption as the other screen? Could we move that to a constant so we can see all the references if/when we replace it?

File: src/Theming/Fluent9DarkTheme.ts [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+    colorPaletteLightGreenBackground1: '#f2fbf2',
+    colorPaletteLightGreenBackground2: '#a7e3a5',
+    colorPaletteLightGreenBackground3: '#13a10e',
+    colorPaletteLightGreenForeground1: '#11910d',
+    colorPaletteLightGreenForeground2: '#0b5a08',
+    colorPaletteLightGreenForeground3: '#13a10e',
+    colorPaletteLightGreenBorderActive: '#13a10e',
+    colorPaletteLightGreenBorder1: '#a7e3a5',
+    colorPaletteLightGreenBorder2: '#13a10e',
+    colorPaletteMarigoldBackground1: '#fefbf4',
```
Comment (msnyder-msft) [inline src/Theming/Fluent9DarkTheme.ts:285]: Maybe we can have defaults in one file and then have our overrides in another variable or file so we know what we have changed from the default styles?

File: src/Theming/FluentThemes.ts [hunk 7]
```diff
@@ -30,6 +37,21 @@ export const getFluentTheme = (theme: Theme): ITheme => {
     }
 };
 
+export const getVNextTheme = (theme: Theme): Fluent9Theme => {
+    switch (theme) {
+        case Theme.Dark:
+            return Fluent9DarkTheme;
+        case Theme.Light:
```
Comment (msnyder-msft) [inline src/Theming/FluentThemes.ts:44]: nit: can remove the light theme and just let it fall through.

**Review by msnyder-msft — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (pasanchMSFT) [reply package.json:23]: generate fluent 9 template components

File: src/Apps/Legion/Components/ConsumeShell/ConsumeShell.tsx (hunk 2 above)
Comment (pasanchMSFT) [reply src/Apps/Legion/Components/ConsumeShell/ConsumeShell.tsx:43]: I tried using the variable directly, but typescript had issues with understanding it was a string. I can try to type it and see if that works.

File: package.json [hunk 8]
```diff
@@ -21,6 +21,7 @@
...
+        "generate-next": "node ./tools/generate-next-template.js",
```
Comment (msnyder-msft) [inline package.json:24]: could we replace `npm run generate` to run this command and rename the existing command to `npm run generate-legacy`?

File: src/Apps/Legion/Components/ConsumeShell/ConsumeShell.types.ts (hunk 3 above)
Comment (pasanchMSFT) [reply src/Apps/Legion/Components/ConsumeShell/ConsumeShell.types.ts:4]: This is the equivalent of `IComponentNameStyles` it gives the style function the keys to the styles you want (i.e., root). `makeStyles<IComponentClassNames>` makes sure you don't miss any class name.

File: src/Theming/Fluent9DarkTheme.ts (hunk 4 above)
Comment (pasanchMSFT) [reply src/Theming/Fluent9DarkTheme.ts:3]: I'll iterate on this, see if I can use Partial<Theme> or something else to just define overrides. Just wanted to push this as a v0.

File: src/Theming/Fluent9ExplorerTheme.ts [hunk 9]
```diff
@@ -0,0 +1,397 @@
+import { Theme } from '@fluentui/react-components';
+
+export const Fluent9ExplorerTheme: Theme = {
+    borderRadiusNone: '0',
```
Comment (iremgokce) [inline src/Theming/Fluent9ExplorerTheme.ts:4]: hmm is there a way to extend these from? Seems like we have the same set of variables both in other theme ts files?

**Review by iremgokce — APPROVED** [review]
I left one comment for extending common css properties from theme files.

File: src/Apps/Legion/Components/ConsumeShell/ConsumeShell.types.ts (hunk 3 above)
Comment (msnyder-msft) [reply src/Apps/Legion/Components/ConsumeShell/ConsumeShell.types.ts:4]: hm ok. I just wasn't seeing where it was used. I must have missed it.

**Pull request merged**